aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/BaseHandlerStd.java

```diff
@@ -460,31 +458,63 @@ protected boolean isDbInstanceDeleted(
             final ProxyClient<RdsClient> rdsProxyClient,
             final ResourceModel model
     ) {
+        DBInstance dbInstance;
         try {
-            fetchDBInstance(rdsProxyClient, model);
+            dbInstance = fetchDBInstance(rdsProxyClient, model);
         } catch (DbInstanceNotFoundException e) {
             // the instance is gone, exactly what we need
             return true;
         }
+
+        assertNoTerminalStatus(dbInstance);
+
         return false;
     }
 
-    protected boolean isDbInstanceStabilized(
+    private void assertNoTerminalStatus(final DBInstance dbInstance) throws CfnNotStabilizedException {
+        final DBInstanceStatus status = DBInstanceStatus.fromString(dbInstance.dbInstanceStatus());
+        if (status != null && status.isTerminal()) {
+            throw new CfnNotStabilizedException(new Exception("DB Instance is in state: " + status.toString()));
```

Comment:
nit: This does not match the rest of the code that we use for returning values to CFN framework. As far as I can tell, throwing `CfnNotStabilizedException` is equivalent of returning failed result with an error code `HandlerErrorCode.NotStabilized`. 

Do you think using the same error handling style is going to introduce more complexity?

Reply:
I totally agree with you: we've been sticking to returning a faulty event back to the handler. The reason this approach is used here is a lack of an interruption mechanism in the stabilizer (it seems):
```java
/**
  * @param callback, the stabilize predicate that is called several times to
  *            determine success.
  * @return true if the condition of stabilization has been meet else false.
  */
```
Thinking of an alternative, we can throw some other exception here and catch it outside the code block that performs stabilization. Having that said, i believe we will have quite some more code blocks wrapped in try-catch blocks that would make it look clumsy. To me this shortcut looks like a fair balance between this bulky try-catch party and an oopsie-doopsie-we-got-another-exit-point situation. Please let me know if it makes sense.

Reply:
Yeah, looks like an oversight in stabilization mechanism. Let's keep CFN exception for now 👍